src/js/components/Box/Box.js

```diff
@@ -223,7 +235,7 @@ const Box = forwardRef(
         aria-label={a11yTitle}
         background={background}
         border={border}
-        ref={ref}
+        ref={typeof as === 'function' ? undefined : containerRef}
```

Comment:
Can you add an inline comment for why this condition is necessary?

Reply:
Certainly. I have changes in my sandbox also that will do a console.warn in this situation (both the pre-existing ref+as function situation and the responsive="container" + as function situation)